pkg/integrations/v2/autoscrape/appender.go

```diff
@@ -0,0 +1,32 @@
+package autoscrape
+
+import (
+	"fmt"
+
+	"github.com/prometheus/prometheus/pkg/exemplar"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+)
+
+// failedAppender is used as the appender when an instance couldn't be found.
+type failedAppender struct {
+	instanceName string
+}
+
```

Comment:
Right, so this is just being used as a safety fallback. Ideally we never hit this point, but if we didn't return _some_ kind of appender, the scraper code would panic. 

I think it might be possible to hit this during a reload, actually:

1. User renames an instance  
2. The metrics subsystem updates with the new instance name 
3. At the same time, an autoscrape happens, still configured with the old name. It fails.
4. Finally, after the metrics subsystem is updated, the integrations subsystem is updated and the autoscraper is configured properly again.